sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionResolver.java

```diff
@@ -93,24 +93,25 @@ static short generateHashCode(String partitionKey) {
      * This implementation is a direct port of the Event Hubs service code; it is intended to match the gateway hashing
      * algorithm as closely as possible and should not be adjusted without careful consideration.
      *
+     * NOTE: Suppressing fallthrough warning for switch-case because we want it to fall into the subsequent cases.
+     *
      * @param data The data to base the hash on.
      * @param seed1 Seed value for the first hash.
      * @param seed2 Seed value for the second hash.
      *
      * @return An object containing the computed hash for {@code seed1} and {@code seed2}.
      */
+    @SuppressWarnings("fallthrough")
     private static Hashed computeHash(byte[] data, int seed1, int seed2) {
-        int a;
-        int b;
-        int c;
-
-        a = b = c = (0xdeadbeef + data.length + seed1);
-        c += seed2;
+        int a = (0xdeadbeef + data.length + seed1);
+        int b = a;
+        int c = a + seed2;
 
         final ByteBuffer buffer = ByteBuffer.allocate(data.length)
-            .put(data)
-            .flip()
-            .order(ByteOrder.LITTLE_ENDIAN);
+            .put(data);
+
+        buffer.flip();
```

Comment:
In Java8, `flip()` method returns a class of type Buffer. But in Java 9+, it returns a type of ByteBuffer. So we split this up to avoid the compilation errors.